src/libraries/System.Security.Cryptography.Encoding/src/System/Security/Cryptography/PemEncoding.cs

```diff
@@ -0,0 +1,532 @@
+                    // 0 or -1 characters, which means we'll probably end up here again,
+                    // advancing 0 or -1 characters, in a loop. To avoid getting stuck,
+                    // detect this situation and return.
+                    fields = default;
+                    return false;
+                }
+                areaOffset += preebEndIndex;
+            }
+
+            fields = default;
+            return false;
+
+            static ReadOnlySpan<char> WritePostEB(ReadOnlySpan<char> label, Span<char> destination)
+            {
+                int size = PostEBPrefix.Length + label.Length + Ending.Length;
+                Debug.Assert(destination.Length >= size);
+                PostEBPrefix.AsSpan().CopyTo(destination);
+                label.CopyTo(destination.Slice(PostEBPrefix.Length));
+                Ending.AsSpan().CopyTo(destination.Slice(PostEBPrefix.Length + label.Length));
+                return destination.Slice(0, size);
+            }
+        }
+
+        private static int IndexOfByOffset(this ReadOnlySpan<char> str, ReadOnlySpan<char> value, int startPosition)
+        {
+            int index = str.Slice(startPosition).IndexOf(value);
+            return index == -1 ? -1 : index + startPosition;
+        }
+
+        private static bool IsValidLabel(ReadOnlySpan<char> data)
+        {
+            static bool IsLabelChar(char c) => (uint)(c - 0x21u) <= 0x5du && c != '-';
+
+            // Empty labels are permitted per RFC 7468.
+            if (data.IsEmpty)
+                return true;
+
+            // First character of label must be a labelchar, which is a character
+            // in 0x21..0x7e (both inclusive), except hyphens.
+            if (!IsLabelChar(data[0]))
+                return false;
+
+            bool previousSpaceOrHyphen = false;
+            for (int index = 1; index < data.Length; index++)
+            {
+                char c = data[index];
+
+                if (IsLabelChar(c))
+                {
+                    previousSpaceOrHyphen = false;
+                    continue;
+                }
+
+                bool isSpaceOrHyphen = c == ' ' || c == '-';
+
+                // IETF RFC 7468 states that every character in a label must
+                // be a labelchar, and each labelchar may have zero or one
+                // preceding space or hyphen, except the first labelchar.
+                // If this character is not a space or hyphen, then this characer
+                // is invalid.
+                // If it is a space or hyphen, and the previous character was
+                // also a space or hyphen, then we have two consecutive spaces
+                // or hyphens which is is invalid.
+                if (!isSpaceOrHyphen || previousSpaceOrHyphen)
+                {
+                    return false;
+                }
+
+                previousSpaceOrHyphen = true;
+            }
+
+            return true;
```

Comment:
A space-or-hyphen has to be followed by another labelchar.  So isn't this `return !previousSpaceOrHyphen;`?

I think right now it'll say "HELLO-" is a valid label. (I don't see a test confirming one way or another)

Reply:
You're correct. I misread the RFC.